embedchain/embedchain.py

```diff
@@ -164,35 +164,45 @@ def get_openai_answer(self, prompt):
         )
         return response["choices"][0]["message"]["content"]
     
-    def retrieve_from_database(self, input_query):
+    def retrieve_from_database(self, input_query, number_documents = 1):
         """
         Queries the vector database based on the given input query.
         Gets relevant doc based on the query
 
         :param input_query: The query to use.
-        :return: The content of the document that matched your query.
+        :number_documents: The number of documents to retrieve from the database.
+        :return: List with the contents of the document that matched your query.
         """
         result = self.collection.query(
             query_texts=[input_query,],
-            n_results=1,
+            n_results=number_documents,
         )
         result_formatted = self._format_result(result)
-        content = result_formatted[0][0].page_content
-        return content
+        print(result_formatted)
+        contents = [document[0].page_content for document in result_formatted]
+        return contents
     
-    def generate_prompt(self, input_query, context):
+    def generate_prompt(self, input_query, contexts):
         """
         Generates a prompt based on the given query and context, ready to be passed to an LLM
 
         :param input_query: The query to use.
-        :param context: Similar documents to the query used as context.
+        :param contexts: List of similar documents to the query used as context.
         :return: The prompt
         """
-        prompt = f"""Use the following pieces of context to answer the query at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.
-        {context}
-        Query: {input_query}
-        Helpful Answer:
-        """
+        if len(contexts) > 0:
```

Comment:
Could you remove this condition? We don't need it.


```python
prompt = f"""Use the following context to answer the query at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.
            {" | ".join(contexts)}
            Query: {input_query}
            Helpful Answer:
            """

```

Reply:
It should be > 1. The prosa text uses the plural then.